framework/lib/theme/components/tag/index.ts

```diff
@@ -28,16 +25,18 @@ export const Tag: ComponentMultiStyleConfig = {
     },
   },
   variants: {
-    solid: ({ bgColor, colorScheme }) => ({
+    solid: ({ theme: { colors }, colorScheme }) => ({
       container: {
-        bgColor: bgColor ?? colorScheme,
-        color: getContrastColor(bgColor ?? useToken('colors', colorScheme)),
+        bgColor: colors[colorScheme] && colors[colorScheme][500]
+          ? colors[colorScheme][500] : colorScheme,
+        color: 'white',
       },
     }),
-    subtle: () => ({
+    subtle: ({ theme: { colors }, colorScheme }) => ({
       container: {
-        bgColor: useToken('colors', 'background.tag.subtle'),
-        color: 'text.default',
+        bgColor: colors[colorScheme] && colors[colorScheme][100]
+          ? colors[colorScheme][100] : colorScheme,
+        color: 'default',
```

Comment:
In most of the cases for text or <span> tags this would have been right, or we would use our tokens which are mapped if the theme changes. So using `text.default` which translates form black to white when the theme changes is a perfect way to handle text over layer elements.

But in this case there is no design request for text change if the theme changes. So surface + text stay the same, so in this case you will need to change this line into `color: 'mono.black',`